modules/telemetry/telemetry/include/hephaestus/telemetry/metric_record.h

```diff
@@ -39,17 +42,21 @@ void record(const Metric& metric);
 /// @param data The data to record, needs to be json serializable.
 /// @param timestamp The timestamp of the metric, if not provided, the current time is used.
 template <typename DataT>
-void record(const std::string& component, const std::string& tag, const DataT& data,
-            ClockT::time_point timestamp = ClockT::now()) {
-  const auto json = serdes::serializeToJSON(data);
-  const Metric metric{
-    .component = component,
-    .tag = tag,
-    .timestamp = timestamp,
-    .values = internal::jsonToValuesMap(json),
-  };
-
-  record(metric);
+void record(std::string component, std::string tag, DataT&& data) {
```

Comment:
is the change of interface necessary here? i don't see why the timestamp interface needs to be removed